### PR TITLE
Leaf 4204 - Setting the 404 on decommissioned sites

### PR DESCRIPTION
**Pull request by jampaul3**
Summary:
Most of the work for this ticket happened on the Leaf-site_specific repo but we did have to account for decommissioned sites in the base code here. There will be some setup in the workflow once this goes out but all the coding is here for it to function.

Potential Impact:
This will allow a leaf coach to decommission a site from the launchpad. The only potential negative impact I can see is the fields in the db don't get added, causing a fatal for a field not being found in the query.

Testing:
Testing of this can be found in the Leaf-site_specific repo PR.

### CODE REVIEW

File: app/Leaf/Site.php [hunk 1]
```diff
@@ -82,7 +82,7 @@ private function checkPath(): array
     {
         $vars = array(':site_path' => $this->match);
         $sql = 'SELECT `site_path`, `site_uploads`, `portal_database`, `orgchart_path`,
-                    `orgchart_database`
+                    `orgchart_database`, `launchpadID`, `decommission`
                 FROM `sites`
```
Comment (mgaoVA) [inline app/Leaf/Site.php:86]: Couple things:
- This highlights another prerequisite that didn't come up before. The "sites" table is apparently missing from the [resources database boilerplate](https://github.com/department-of-veterans-affairs/LEAF/blob/master/docker/mysql/db/boilerplate/resource_database_boilerplate.sql), and should be added. All Request Portals should have identical **core** codebases and database schemas.
- LEAF-4204 outlines a specific naming convention for the new column: "decommissionTimestamp". This should not be changed unless there's a compelling reason.

File: app/libs/loaders/Leaf_autoloader.php [hunk 2]
```diff
@@ -23,12 +23,17 @@
 $site = new App\Leaf\Site($file_paths_db, $_SERVER['SCRIPT_FILENAME']);
 
 if ($site->error) {
-    throw new Exception("Sorry the page you are looking for could not be found, please check the url and try again.");
```
Comment (mgaoVA) [inline app/libs/loaders/Leaf_autoloader.php:26]: Note that http_response_code() does not abort processing, and removing the throw substantially changes program flow. Not sure if this was intended.

**Review by mgaoVA — CHANGES_REQUESTED** [review]

File: app/libs/loaders/Leaf_autoloader.php (hunk 2 above)
Comment (jampaul3) [reply app/libs/loaders/Leaf_autoloader.php:26]: When I tested the throw on staging it threw a 500 with no real message, that is why I changed it to this and added the die() right after it. This way the nginx will capture the 400 and send them to a 4xx page.

File: app/Leaf/Site.php (hunk 1 above)
Comment (jampaul3) [reply app/Leaf/Site.php:86]: Ok, these have been addressed and will be pushed soon.

**Review by shaneodd — APPROVED** [review]

**Review by aerinkayne — APPROVED** [review]

**Review by Pelentan — APPROVED** [review]

**Review by Pelentan — APPROVED** [review]

**Pull request merged**